### PR TITLE
Refactor: rename variable names (InGameLobbyWatcher.java)

### PR DESCRIPTION
**Pull request by DanVanAtta**

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java [hunk 1]
```diff
@@ -218,11 +218,13 @@ public static InGameLobbyWatcher newInGameLobbyWatcher(
...
               mac,
               login);
-      final UnifiedMessenger um = new UnifiedMessenger(messenger);
-      final RemoteMessenger rm = new RemoteMessenger(um);
-      final RemoteHostUtils rhu = new RemoteHostUtils(messenger.getServerNode(), gameMessenger);
-      rm.registerRemote(rhu, IRemoteHostUtils.Companion.newRemoteNameForNode(um.getLocalNode()));
-      return new InGameLobbyWatcher(messenger, rm, gameMessenger, handler, oldWatcher);
+      final var unifiedMessenger = new UnifiedMessenger(messenger);
+      final var remoteMessenger = new RemoteMessenger(unifiedMessenger);
+      final var remoteHostUtils = new RemoteHostUtils(messenger.getServerNode(), gameMessenger);
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java]: Excellent use of the var keyword 👍

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java [hunk 2]
```diff
@@ -302,10 +304,10 @@ private void postUpdate(final GameDescription newDescription) {
 
   void shutDown() {
     isShutdown = true;
-    if (messenger instanceof IClientMessenger) {
-      ((IClientMessenger) this.messenger).removeErrorListener(messengerErrorListener);
+    if (lobbyMessenger instanceof IClientMessenger) {
+      ((IClientMessenger) this.lobbyMessenger).removeErrorListener(messengerErrorListener);
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java]: I wonder if we could move listener cleanup to shutdown. On IClientMessenger implementations

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java]: Thanks! It's not universally a good thing, but this does seem to be a really good example of where it's useful.

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java (hunk 2 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java]: A lot of the listener cleanup is very suspeicious IMO. I suspect we might be able to further and not need the cleanup and instead rely on tight/proper scoping. With that said, moving the cleanup code could be a good first step.

**Pull request merged**